contents/blog/version-30-upgrades/index.mdx

```diff
@@ -0,0 +1,80 @@
+---
+title: Marquez 30.0 offers a big performance bump, with some assembly required
+date: 2023-02-02
+author: Michael Robinson
+image: ../column-lineage-demo/image.svg
+banner: ../column-lineage-demo/banner.svg
+description: How to ensure a smooth transition to the latest version of Marquez. 
+---
+
+## Marquez 30.0
+
+[Marquez 30.0](https://github.com/MarquezProject/marquez/releases/tag/0.30.0), 
+which we released on January 31, contains some exciting new features, including 
+[column-level lineage](https://github.com/MarquezProject/marquez/pull/2293) and 
+a [soft delete option](https://github.com/MarquezProject/marquez/pull/2343) in 
+the UI.
+
+Along with these long-awaited UI upgrades come some significant performance 
+improvements to the API. In prior versions of Marquez, accessing OpenLineage 
+facets required direct queries against the OpenLineage events table, which could 
+be quite costly depending on the size of this table. Some users have events 
+tables with more than 100K rows. To address this performance issue, Marquez 30.0 
+has new tables for dataset, job and run facets.
+
+So far, so simple, right? Protecting against data loss and ensuring a smooth 
+experience for users was not so simple, in fact. We needed a way to migrate the 
+data while not only avoiding loss but also keeping downtime to a minimum. The 
+solution that [Willy](https://github.com/wslulciuc) and 
+[Pawel](https://github.com/pawel-big-lebowski) found (with input from 
+[Mike Collado](https://github.com/collado-mike)) was to keep the data in the 
+events table until the migration process is complete and to create views on top 
+of the table for accessing the facets. After migration is complete, the views 
+are replaced to point to the new tables. Using views provides safeguards and 
+minimizes downtime during the migration. 
+
+## Migration Procedure
+
+As the 
+[migration plan](https://github.com/MarquezProject/marquez/blob/main/api/src/main/resources/marquez/db/migration/V57__readme.md) 
+says, users whose `lineage_events` tables have up to 100K rows can set it and 
+forget it -- no further action is required. A standard Flyway migration will 
```

Comment:
I think it would be good to be specific here - no further action beyond just upgrading and restarting Marquez? I think that's accurate, but I had some doubts about whether there are additional steps.

The migration guide is also unclear to me. It reads:

> A standard Flyway migration is run which fills newly created tables into dataset_facets, run_facets and job_facets. No extra work is required but be prepared for a couple of minutes downtime when performing upgrade.

The "a migration is run" wording suggests that the user is passive here. But "when performing upgrade" suggests that the user performs the upgrade.

Reply:
Thanks @rossturk . I'll add the phrase about upgrading/restarting. Would also appreciate clarification of whether or not the user has to do anything. Hope someone will chime in.

Reply:
If this migration for <100k rows happens automatically when Marquez starts (which seems the likely case), I propose the following wording:

> no extra work is required beyond upgrading and restarting Marquez, but be prepared...

And the following in the migration guide:

> "A standard Flyway migration is run _upon startup_ which....be prepared for a couple of minutes _of_ downtime _while the migration is performed_"